nemo/collections/asr/modules/flashlight_decoder.py

```diff
@@ -0,0 +1,253 @@
+                        continue
+                    self.trie.insert(spelling_idxs, word_idx, score)
+            self.trie.smear(SmearingMode.MAX)
+    
+            self.decoder_opts = LexiconDecoderOptions(
+                beam_size=beam_size,
+                beam_size_token=int(beam_size_token),
+                beam_threshold=beam_threshold,
+                lm_weight=lm_weight,
+                word_score=word_score,
+                unk_score=unk_weight,
+                sil_score=sil_weight,
+                log_add=False,
+                criterion_type=self.criterion_type,
+            )
+            
+            self.decoder = LexiconDecoder(
+                self.decoder_opts,
+                self.trie,
+                self.lm,
+                self.silence,
+                self.blank,
+                self.unk_word,
+                [],
+                self.unit_lm,
+            )
+        else:
+            assert self.unit_lm, "lexicon free decoding can only be done with a unit language model"
+            from flashlight.lib.text.decoder import LexiconFreeDecoder, LexiconFreeDecoderOptions
+
+            d = {w: [[w]] for w in self.tokenizer.vocab + ([] if '<unk>' in self.tokenizer.vocab else ['<unk>'])}
+            self.word_dict = create_word_dict(d)
+            self.lm = KenLM(lm_path, self.word_dict)
+            self.decoder_opts = LexiconFreeDecoderOptions(
+                beam_size=beam_size,
+                beam_size_token=int(beam_size_token),
+                beam_threshold=beam_threshold,
+                lm_weight=lm_weight,
+                sil_score=sil_weight,
+                log_add=False,
+                criterion_type=self.criterion_type,
+            )
+            self.decoder = LexiconFreeDecoder(
+                self.decoder_opts, self.lm, self.silence, self.blank, []
+            )
+    
+    def _get_tokens(self, idxs: List[int]):
+        """Normalize tokens by handling CTC blank, ASG replabels, etc."""
+        
+        idxs = (g[0] for g in it.groupby(idxs))
+        idxs = filter(lambda x: x != self.blank and x != self.silence, idxs)
+        
+        return torch.LongTensor(list(idxs))
+    
+    def _get_timesteps(self, token_idxs: List[int]):
+        """Returns frame numbers corresponding to every non-blank token.
+        Parameters
+        ----------
+        token_idxs : List[int]
+            IDs of decoded tokens.
+        Returns
+        -------
+        List[int]
+            Frame numbers corresponding to every non-blank token.
+        """
+        
+        timesteps = []
+        for i, token_idx in enumerate(token_idxs):
+            if token_idx == self.blank:
+                continue
+            if i == 0 or token_idx != token_idxs[i-1]:
+                timesteps.append(i)
+        
+        return timesteps
+
+    #@typecheck(ignore_collections=True)
+    @torch.no_grad()

```

Comment:
Can you add back the types if possible? if not, its fine to keep them commented for now.

Reply:
It's not possible as we discussed in our meeting, will leave them uncommented for now